entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java

```diff
@@ -175,13 +181,21 @@ public boolean unjoinCurrentUserFromSite(EntityView view, Map<String, Object> pa
         } else if ("site".equals(siteId)) {
             siteId = view.getPathSegment(3);
         }
+
         if (siteId == null) {
             throw new IllegalArgumentException(
                     "siteId must be set in order to unjoin sites, set in params or in the URL /unjoin/site/siteId");
         }
         checkSiteSecurity(siteId);
         try {
             siteService.unjoin(siteId);
+            //String user = sessionManager().getCurrentSessionUserId();
+            String currentUserEid = userEntityProvider.getCurrentUser(view).getEid(); //userDirectoryService.getCurrentUser().getEid();
+            String roleId = siteService.getSite(siteId).getJoinerRole();
+            String[] userAuditString = {siteId,currentUserEid,roleId,UserAuditService.USER_AUDIT_ACTION_REMOVE,userAuditRegistration.getDatabaseSourceKey(),currentUserEid};
+            List<String[]> userAuditList = new ArrayList<>();
```

Comment:
These two lines can be combined; `Arrays.asList....`

Also, is there a reason you're creating a List of String arrays? Why not `List<List<String>>`?